RCTAppleHealthKit/RCTAppleHealthKit.m

```diff
@@ -329,4 +353,99 @@ - (void)getModuleInfo:(NSDictionary *)input callback:(RCTResponseSenderBlock)cal
     callback(@[[NSNull null], info]);
 }
 
+#pragma mark Meals
+-(void)reportHealthKitMeal:(NSDictionary *)input resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject
+{
+    NSString *id = [input objectForKey:@"id"];
+    NSString *name = [input objectForKey:@"name"];
+    NSString *dateTimeString = [input objectForKey:@"dateTime"];
+    NSString *type = [input objectForKey:@"type"];
+    int energy = [[input valueForKey:@"energy"] integerValue];
+    NSDictionary *macros = [input objectForKey:@"macros"];
```

Comment:
one thing about these which is different from the swift counterpart, is that we need to verify that they are also non-nil values. (Or check the documentation on HealthKit to see if `id`, `name`, `dateTimeString`, `type` can be nil. Also, on the swift counter part, if `macros` was nil it would fail...


But we also have some reported bugs about healthkit integration not working properly

Reply:
`HealthKit` throws an error if invalid data provided, I think we just can catch it there

Reply:
in the case that `macros` is nil, we'll get 0 for each of those values, so I'd actually be inclined to bail early here as well, what do you think?